<launch>
  <master auto="start"/>
  <param name="/use_sim_time" value="false"/>

  <arg name="image_debug" default="false" />
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg     if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />

  <include file="$(find sbpl_perception)/config/ycb_objects.xml"/>

  <node pkg="object_recognition_node" type="object_localizer" name="object_localizer_service" output="screen" launch-prefix="/media/aditya/A69AFABA9AFA85D9/Cruzr/code/openmpi-4.0.0/install/bin/mpirun -n 4 --mca mpi_yield_when_idle 1" respawn="false">
      <rosparam command="load" file="$(find sbpl_perception)/config/pr2_env_config.yaml" />
      <rosparam command="load" file="$(find sbpl_perception)/config/pr2_planner_config.yaml" />
      <rosparam command="load" file="$(find sbpl_perception)/config/camera_config.yaml" />
      <param name="image_debug" value="$(arg image_debug)"/>
  </node>

  <node pkg="object_recognition_node" type="obj_rec_node" name="obj_rec_node" output="screen" launch-prefix="$(arg launch_prefix)" respawn="false">
    <!-- <remap from="/keypress_topic" to="/obj_recognition_keyboard_listener/keydown"/> -->
    <remap from="input_cloud" to="/camera/depth_registered/points"/>
    <param name="pcl_visualization" value="true"/>
    <!-- <param name="table_height" value="1.127"/> <!&#45;&#45;1.13&#45;&#45;> -->
    <param name="table_height" value="0.535"/> <!--1.13-->
    <param name="xmin" value="0.0"/> <!--1.13-->
    <param name="xmax" value="2.2"/> <!--1.13-->
    <param name="ymin" value="-1.4"/> <!--1.13-->
    <param name="ymax" value="1.4"/> <!--1.13-->
    <param name="reference_frame" value="/world"/>
    <param name="camera_frame" value="/camera_rgb_optical_frame"/>
  </node>

  <!-- <node pkg="keyboard" type="keyboard" name="obj_recognition_keyboard_listener" output="screen"> -->
  <!-- </node> -->

</launch>
